gui/src/chartdbs.cpp

```diff
@@ -1524,15 +1551,15 @@ bool ChartDatabase::Update(ArrayOfCDI &dir_array, bool bForce,
 #endif
 
     wxString dir_magic;
-
-    if (dir_info.fullpath.Find(_T("GSHHG")) != wxNOT_FOUND) {
-      if (!wxDir::FindFirst(dir_info.fullpath, "poly-*-1.dat").empty()) {
-        // If some polygons exist in the directory, set it as the one to use for
-        // GSHHG
-        // TODO: We should probably compare the version and maybe resolutions
-        // available with what is currently used...
-        gWorldMapLocation = dir_info.fullpath + wxFileName::GetPathSeparator();
-      }
+    // Recursively search for a directory that contains GSHHG files starting from dir_info.
+    wxString gshhg_dir = findGshhgDirectory(dir_info.fullpath);
+    if (!gshhg_dir.empty()) {
+      // If some polygons exist in the directory, set it as the one to use for
+      // GSHHG
+      // TODO: We should probably compare the version and maybe resolutions
+      // available with what is currently used...
```

Comment:
This comment makes sense, but I feel it should be done in a separate PR. I also see the suggestion about not using the GSHHG files anymore.